cordova-lib/src/cordova/platform.js

```diff
@@ -110,15 +110,25 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                 if(platform === 'wp8') {
                     events.emit('warn', 'wp8 has been deprecated. Please use windows instead.');
                 }
-                if(spec && pkgJson) {
-                    pkgJson.dependencies[pkgJsonSpec] = spec;
+                // If there is a spec specified during add, add to pkg.json dependencies(if exists).
```

Comment:
Added 113 to 120 because if you are using the specified spec, but do not update and write in pkg.json- when restore happens, it will restore the wrong version. What do you think about this @stevengill ?